// rmc-picker/es/MultiPicker.native.js
import _extends from 'babel-runtime/helpers/extends';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React from 'react';
import { View, StyleSheet } from 'react-native';
import reactMixin from 'react-mixin';
import Picker from './Picker';
import MultiPickerMixin from './MultiPickerMixin';
var styles = StyleSheet.create({
    item: {
        flex: 1
    },
    root: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingTop: 10,
        paddingBottom: 10
    }
});

var MultiPicker = function (_React$Component) {
    _inherits(MultiPicker, _React$Component);

    function MultiPicker() {
        _classCallCheck(this, MultiPicker);

        return _possibleConstructorReturn(this, (MultiPicker.__proto__ || Object.getPrototypeOf(MultiPicker)).apply(this, arguments));
    }

    _createClass(MultiPicker, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            var props = this.props;
            var disabled = props.disabled,
                pickerItemStyle = props.pickerItemStyle,
                pure = props.pure,
                children = props.children,
                style = props.style;

            var selectedValue = this.getValue();
            var colElements = children.map(function (col, i) {
                return React.createElement(
                    View,
                    { key: col.key || i, style: styles.item },
                    React.createElement(Picker, _extends({ itemStyle: pickerItemStyle, disabled: disabled, pure: pure, selectedValue: selectedValue[i], onValueChange: _this2.onValueChange.bind(_this2, i) }, col.props))
                );
            });
            return React.createElement(
                View,
                { style: [styles.root, style] },
                colElements
            );
        }
    }]);

    return MultiPicker;
}(React.Component);

reactMixin.onClass(MultiPicker, MultiPickerMixin);
export default MultiPicker;
