// rmc-picker/es/NativePicker.android.js
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React from 'react';
import { ScrollView, View, StyleSheet, PixelRatio, Text } from 'react-native';
import reactMixin from 'react-mixin';
import PickerMixin from './PickerMixin';
var ratio = PixelRatio.get();
var styles = StyleSheet.create({
    indicator: {
        position: 'absolute',
        left: 0,
        top: -99,
        borderColor: '#aaa',
        borderTopWidth: 1 / ratio,
        borderBottomWidth: 1 / ratio
    },
    scrollView: {
        height: 0
    },
    selectedItemText: {
        fontSize: 21,
        fontWeight: 'bold',
        color: '#000'
    },
    itemText: {
        fontSize: 20,
        color: '#aaa',
        textAlign: 'center'
    }
});

var Picker = function (_React$Component) {
    _inherits(Picker, _React$Component);

    function Picker() {
        _classCallCheck(this, Picker);

        var _this = _possibleConstructorReturn(this, (Picker.__proto__ || Object.getPrototypeOf(Picker)).apply(this, arguments));

        _this.onItemLayout = function (e) {
            var _e$nativeEvent$layout = e.nativeEvent.layout,
                height = _e$nativeEvent$layout.height,
                width = _e$nativeEvent$layout.width;
            // console.log('onItemLayout', height);

            if (_this.itemHeight !== height || _this.itemWidth !== width) {
                _this.itemWidth = width;
                _this.refs.indicator.setNativeProps({
                    style: [styles.indicator, {
                        top: height * 3,
                        height: height,
                        width: width
                    }]
                });
            }
            if (_this.itemHeight !== height) {
                _this.itemHeight = height;
                _this.refs.scroller.setNativeProps({
                    style: {
                        height: height * 7
                    }
                });
                _this.refs.content.setNativeProps({
                    style: {
                        paddingTop: height * 3,
                        paddingBottom: height * 3
                    }
                });
                // i do no know why!...
                setTimeout(function () {
                    _this.select(_this.props.selectedValue);
                }, 0);
            }
        };
        _this.onScroll = function (e) {
            var y = e.nativeEvent.contentOffset.y;

            _this.clearScrollBuffer();
            _this.scrollBuffer = setTimeout(function () {
                _this.clearScrollBuffer();
                _this.doScrollingComplete(y);
            }, 100);
        };
        return _this;
    }

    _createClass(Picker, [{
        key: 'componentDidUpdate',
        value: function componentDidUpdate() {
            this.select(this.props.selectedValue);
        }
    }, {
        key: 'componentWillUnMount',
        value: function componentWillUnMount() {
            this.clearScrollBuffer();
        }
    }, {
        key: 'clearScrollBuffer',
        value: function clearScrollBuffer() {
            if (this.scrollBuffer) {
                clearTimeout(this.scrollBuffer);
            }
        }
    }, {
        key: 'scrollTo',
        value: function scrollTo(y) {
            this.refs.scroller.scrollTo({
                y: y,
                animated: false
            });
        }
    }, {
        key: 'fireValueChange',
        value: function fireValueChange(selectedValue) {
            if (this.props.selectedValue !== selectedValue && this.props.onValueChange) {
                this.props.onValueChange(selectedValue);
            }
        }
    }, {
        key: 'getChildMember',
        value: function getChildMember(child, m) {
            return child.props[m];
        }
    }, {
        key: 'toChildrenArray',
        value: function toChildrenArray(children) {
            return React.Children.toArray(children);
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var _props = this.props,
                children = _props.children,
                itemStyle = _props.itemStyle,
                selectedValue = _props.selectedValue,
                style = _props.style;

            var items = React.Children.map(children, function (item, index) {
                var totalStyle = [styles.itemText];
                if (selectedValue === item.props.value) {
                    totalStyle.push(styles.selectedItemText);
                }
                totalStyle.push(itemStyle);
                return React.createElement(
                    View,
                    { ref: 'item' + index, onLayout: index === 0 ? _this2.onItemLayout : undefined, key: item.key },
                    React.createElement(
                        Text,
                        { style: totalStyle, numberOfLines: 1 },
                        item.props.label
                    )
                );
            });
            return React.createElement(
                View,
                { style: style },
                React.createElement(
                    ScrollView,
                    { style: styles.scrollView, ref: 'scroller', onScroll: this.onScroll, scrollEventThrottle: 16, showsVerticalScrollIndicator: false },
                    React.createElement(
                        View,
                        { ref: 'content' },
                        items
                    )
                ),
                React.createElement(View, { ref: 'indicator', style: styles.indicator })
            );
        }
    }]);

    return Picker;
}(React.Component);

Picker.defaultProps = {
    onValueChange: function onValueChange() {}
};
Picker.Item = function Item() {};
reactMixin.onClass(Picker, PickerMixin);
export default Picker;
