// rmc-picker/es/Picker.js
import _defineProperty from 'babel-runtime/helpers/defineProperty';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React from 'react';
import reactMixin from 'react-mixin';
import classNames from 'classnames';
import ZScroller from 'zscroller';
import PickerMixin from './PickerMixin';
import isChildrenEqual from './isChildrenEqual';

var Picker = function (_React$Component) {
    _inherits(Picker, _React$Component);

    function Picker(props) {
        _classCallCheck(this, Picker);

        var _this = _possibleConstructorReturn(this, (Picker.__proto__ || Object.getPrototypeOf(Picker)).call(this, props));

        _this.scrollingComplete = function () {
            var _this$zscroller$scrol = _this.zscroller.scroller.getValues(),
                top = _this$zscroller$scrol.top;

            if (top >= 0) {
                _this.doScrollingComplete(top);
            }
        };
        var selectedValueState = void 0;
        var selectedValue = props.selectedValue,
            defaultSelectedValue = props.defaultSelectedValue,
            children = props.children;

        if (selectedValue !== undefined) {
            selectedValueState = selectedValue;
        } else if (defaultSelectedValue !== undefined) {
            selectedValueState = defaultSelectedValue;
        } else if (children && children.length) {
            selectedValueState = children[0].value;
        }
        _this.state = {
            selectedValue: selectedValueState
        };
        return _this;
    }

    _createClass(Picker, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            // https://github.com/react-component/m-picker/issues/18
            this.itemHeight = this.refs.indicator.getBoundingClientRect().height;
            // compact
            this.refs.content.style.padding = this.itemHeight * 3 + 'px 0';
            this.zscroller = new ZScroller(this.refs.content, {
                scrollingX: false,
                snapping: true,
                locking: false,
                penetrationDeceleration: .1,
                minVelocityToKeepDecelerating: 0.5,
                scrollingComplete: this.scrollingComplete
            });
            this.zscroller.setDisabled(this.props.disabled);
            this.zscroller.scroller.setSnapSize(0, this.itemHeight);
            this.select(this.state.selectedValue);
        }
    }, {
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            if ('selectedValue' in nextProps) {
                this.setState({
                    selectedValue: nextProps.selectedValue
                });
            }
            this.zscroller.setDisabled(nextProps.disabled);
        }
    }, {
        key: 'shouldComponentUpdate',
        value: function shouldComponentUpdate(nextProps, nextState) {
            return this.state.selectedValue !== nextState.selectedValue || !isChildrenEqual(this.props.children, nextProps.children, this.props.pure);
        }
    }, {
        key: 'componentDidUpdate',
        value: function componentDidUpdate() {
            this.zscroller.reflow();
            this.select(this.state.selectedValue);
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            this.zscroller.destroy();
        }
    }, {
        key: 'scrollTo',
        value: function scrollTo(top) {
            this.zscroller.scroller.scrollTo(0, top);
        }
    }, {
        key: 'fireValueChange',
        value: function fireValueChange(selectedValue) {
            if (selectedValue !== this.state.selectedValue) {
                if (!('selectedValue' in this.props)) {
                    this.setState({
                        selectedValue: selectedValue
                    });
                }
                if (this.props.onValueChange) {
                    this.props.onValueChange(selectedValue);
                }
            }
        }
    }, {
        key: 'getChildMember',
        value: function getChildMember(child, m) {
            return child[m];
        }
    }, {
        key: 'getValue',
        value: function getValue() {
            return this.props.selectedValue || this.props.children && this.props.children[0] && this.props.children[0].value;
        }
    }, {
        key: 'toChildrenArray',
        value: function toChildrenArray(children) {
            return children;
        }
    }, {
        key: 'render',
        value: function render() {
            var _pickerCls;

            var _props = this.props,
                children = _props.children,
                prefixCls = _props.prefixCls,
                className = _props.className,
                itemStyle = _props.itemStyle,
                indicatorStyle = _props.indicatorStyle;
            var selectedValue = this.state.selectedValue;

            var itemClassName = prefixCls + '-item';
            var selectedItemClassName = itemClassName + ' ' + prefixCls + '-item-selected';
            var items = children.map(function (item) {
                return React.createElement(
                    'div',
                    { style: itemStyle, className: selectedValue === item.value ? selectedItemClassName : itemClassName, key: item.value },
                    item.label
                );
            });
            var pickerCls = (_pickerCls = {}, _defineProperty(_pickerCls, className, !!className), _defineProperty(_pickerCls, prefixCls, true), _pickerCls);
            return React.createElement(
                'div',
                { className: classNames(pickerCls) },
                React.createElement('div', { className: prefixCls + '-mask' }),
                React.createElement('div', { className: prefixCls + '-indicator', ref: 'indicator', style: indicatorStyle }),
                React.createElement(
                    'div',
                    { className: prefixCls + '-content', ref: 'content' },
                    items
                )
            );
        }
    }]);

    return Picker;
}(React.Component);

Picker.defaultProps = {
    prefixCls: 'rmc-picker',
    pure: true,
    onValueChange: function onValueChange() {}
};
reactMixin.onClass(Picker, PickerMixin);
export default Picker;
