// rmc-picker/lib/MultiPicker.native.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactNative = require('react-native');

var _reactMixin = require('react-mixin');

var _reactMixin2 = _interopRequireDefault(_reactMixin);

var _Picker = require('./Picker');

var _Picker2 = _interopRequireDefault(_Picker);

var _MultiPickerMixin = require('./MultiPickerMixin');

var _MultiPickerMixin2 = _interopRequireDefault(_MultiPickerMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var styles = _reactNative.StyleSheet.create({
    item: {
        flex: 1
    },
    root: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingTop: 10,
        paddingBottom: 10
    }
});

var MultiPicker = function (_React$Component) {
    (0, _inherits3['default'])(MultiPicker, _React$Component);

    function MultiPicker() {
        (0, _classCallCheck3['default'])(this, MultiPicker);
        return (0, _possibleConstructorReturn3['default'])(this, (MultiPicker.__proto__ || Object.getPrototypeOf(MultiPicker)).apply(this, arguments));
    }

    (0, _createClass3['default'])(MultiPicker, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            var props = this.props;
            var disabled = props.disabled,
                pickerItemStyle = props.pickerItemStyle,
                pure = props.pure,
                children = props.children,
                style = props.style;

            var selectedValue = this.getValue();
            var colElements = children.map(function (col, i) {
                return _react2['default'].createElement(
                    _reactNative.View,
                    { key: col.key || i, style: styles.item },
                    _react2['default'].createElement(_Picker2['default'], (0, _extends3['default'])({ itemStyle: pickerItemStyle, disabled: disabled, pure: pure, selectedValue: selectedValue[i], onValueChange: _this2.onValueChange.bind(_this2, i) }, col.props))
                );
            });
            return _react2['default'].createElement(
                _reactNative.View,
                { style: [styles.root, style] },
                colElements
            );
        }
    }]);
    return MultiPicker;
}(_react2['default'].Component);

_reactMixin2['default'].onClass(MultiPicker, _MultiPickerMixin2['default']);
exports['default'] = MultiPicker;
module.exports = exports['default'];
