// rmc-date-picker/es/Popup.js
var __assign = this && this.__assign || Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
    }
    return t;
};
import React from 'react';
import PopupPicker from 'rmc-picker/es/Popup';
var PopupDatePicker = React.createClass({
    displayName: 'PopupDatePicker',
    getDefaultProps: function getDefaultProps() {
        return {
            pickerValueProp: 'date',
            pickerValueChangeProp: 'onDateChange'
        };
    },
    onOk: function onOk(v) {
        var _props = this.props,
            onChange = _props.onChange,
            onOk = _props.onOk;

        if (onChange) {
            onChange(v);
        }
        if (onOk) {
            onOk(v);
        }
    },
    render: function render() {
        return React.createElement(PopupPicker, __assign({ picker: this.props.datePicker, value: this.props.date }, this.props, { onOk: this.onOk }));
    }
});
export default PopupDatePicker;
